src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Main {
+    private static FigureSupplier figureSupplier = new FigureSupplier();
+    private static ColorSupplier colorSupplier = new ColorSupplier();
+    private static Random random = new Random();
+
+    public static void main(String[] args) {
+        FigureForm[] array = new FigureForm[5];
+        for (int i = 0; i < array.length / 2; i++) {
+            array[i] = figureSupplier.getRandomFigure(random.nextInt(5));
+            if (array[i].equals(FigureSupplier.circle)) {
+                FigureSupplier.circle.draw();
+            }
+            if (array[i].equals(FigureSupplier.isoscelesTrapezoid)) {
+                FigureSupplier.isoscelesTrapezoid.draw();
+            }
+            if (array[i].equals(FigureSupplier.rectangle)) {
+                FigureSupplier.rectangle.draw();
+            }
+            if (array[i] == FigureSupplier.rightTriangle) {
+                FigureSupplier.rightTriangle.draw();
+            }
+            if (array[i] == FigureSupplier.square) {
+                FigureSupplier.square.draw();
+            }
+        }
+        for (int j = array.length / 2; j < array.length; j++) {
+            Circle circle1 = new Circle(10, "WHITE");
+            circle1.draw();
+        }
+        System.out.println();
```

Comment:
No need to write such complex logic
Here, one variable of the FigureSupplier type is required, and in the cycle call the method (getRandomFigure()) of this variable several times
```suggestion
```

Reply:
+